client/js/main.ts

```diff
@@ -391,6 +381,19 @@ const setConfigFromUrl = (map: MandelbrotMap) => {
     }
   }
 };
+const shareButton = document.getElementById(
+  "share-button"
+) as HTMLButtonElement;
+const updateShareButtonText = (text: string) => {
+  shareButton.innerText = text;
+};
+shareButton.onclick = () => {
+  const url = new URL(window.location.href);
```

Comment:
Update this to set the required URL parameters

Reply:
Hi @rosslh, Is this correct way to set the url parameters in main.ts 
```

const shareButton = document.getElementById("share-button") as HTMLButtonElement;
const updateShareButtonText = (text: string) => {
  shareButton.innerText = text;
};
shareButton.onclick = () => {
    const re = config.re.toString(); 
    const im = config.im.toString();
    const zoomLevel = config.zoom.toString();
    const iterations = config.iterations.toString();
    const exponent = config.exponent.toString();
    const colorScheme = config.colorScheme.toString();
    const reverseColors = config.reverseColors.toString();

    const url = new URL(window.location.href);
    url.searchParams.set("re", re);
    url.searchParams.set("im", im);
    url.searchParams.set("z", zoomLevel);
    url.searchParams.set("i", iterations);
    url.searchParams.set("e", exponent);
    url.searchParams.set("c", colorScheme);
    url.searchParams.set("r", reverseColors);

    
  navigator.clipboard.writeText(url.toString()).then(() => {
      updateShareButtonText("Copied link!");
      setTimeout(() => updateShareButtonText("Share this view"), 5000);
    })
};
```

Reply:
Hey @shubhankar-shandilya-india! That works but is pretty verbose. I think it would be simpler to do something like this:

```typescript
const url = `https://mandelbrot.site/?re=${re}&im=${im}&z=${position.z}&i=${config.iterations}&e=${config.exponent}&c=${config.colorScheme}&r=${config.reverseColors}`;

navigator.clipboard.writeText(url).then(() => {
  updateShareButtonText("Copied link!");
  setTimeout(() => updateShareButtonText("Share this view"), 5000);
})
```

Reply:
Hi @rosslh, I got your point, Is this correct in main.ts 
```typescript
const shareButton = document.getElementById("share-button") as HTMLButtonElement;

const updateShareButtonText = (text: string) => {
  shareButton.innerText = text;
};

shareButton.onclick = () => {
    const newUrl = new URL(window.location.href);
    const url = `${newUrl}?re=${config.re}&im=${config.im}&z=${config.zoom}&i=${config.iterations}&e=${config.exponent}&c=${config.colorScheme}&r=${config.reverseColors}`;

    navigator.clipboard.writeText(url).then(() => {
        updateShareButtonText("Copied link!");
        setTimeout(() => updateShareButtonText("Share this view"), 5000);
    });
};
```